app/api/subscriptions.py

```diff
@@ -43,23 +20,27 @@ def __init__(self):
                                     help='No feed ID provided')
         super(SubscriptionListAPI, self).__init__()
 
+    @login_required
     def get(self):
-        return {'subscriptions': [marshal(subscription, subscription_fields) for subscription in user_subscriptions()]}
+        return {'subscriptions': [marshal(subscription, subscription_fields) for subscription in current_user.subscribed]}
 
+    @login_required
     def post(self):
         args = self.reqparse.parse_args()
 
         try:
-            feed_id = int(args['feed_uri'].strip('/').split('/')[-1])
+            feed = Feed.query.get(int(args['feed_uri'].split('/')[-1]))
         except:
             abort(400)
 
-        subscription = {
-            'user_id': logged_in_user_id,
-            'feed_id': feed_id,
-            'mergeable': True
-        }
-        subscriptions.append(subscription)
+        if not feed:
+            abort(400)
```

Comment:
Yeah. The try is in case
```Python 
args['feed_uri'].split('/')[-1]
```
is not an int. If the id simply doesn't exist in the DB, then it will return None instead of raising an exception.

I will specify the exception to catch on ValueError.